docs/nlu/using-bert.rst

```diff
@@ -0,0 +1,185 @@
+
+Using BERT inside Rasa NLU
+==========================
+
+Since Rasa 1.8.0, you can use pre-trained embeddings from language models like
+`BERT <https://arxiv.org/abs/1810.04805>`_ inside of Rasa NLU pipelines.
+The goal of this document is to show how you can do that
+as well as some practical tips on exploring these new tools.
+
+.. contents::
+   :local:
+
+.. _using_bert:
+
+
+.. edit-link::
+
+Setup
+-----
+
+To demonstrate how to use ``BERT`` we will train two pipelines on Sara,
+the demo bot at Rasa. In doing this we will also be able to measure
+the pros and cons of having ``BERT`` in your pipeline.
+
+If you want to reproduce the results in this document you will need 
+to first clone the repository found here:
+
+.. code-block:: bash
+
+    git clone git@github.com:RasaHQ/rasa-demo.git
+
+Once cloned you can install the requirements. Be sure that 
+you explicitly install the ``transformers`` dependency.
+
+.. code-block:: bash
+
+    pip install "rasa[transformers]"
+
+You should now be all set to train an assistant that will
+use ``BERT``. So let's write configuration files that will allow
+us to compare approaches. We'll make a seperate folder 
+where we can place two new configuration files. 
+
+.. code-block:: bash
+
+    mkdir config
+
+For the next step we've created two configuration files. They only
+contain the pipeline part that is relevant for `NLU` model training and hence don't declare any dialogue policies.
+
+**config/config-light.yml**
+
+.. code-block:: yaml
+
+    language: en
+    pipeline:
+    - name: WhitespaceTokenizer
+    - name: CountVectorsFeaturizer
+    - name: CountVectorsFeaturizer
+    analyzer: char_wb
+    min_ngram: 1
+    max_ngram: 4
+    - name: DIETClassifier
+    epochs: 50
+
+**config/config-heavy.yml**
+
+.. code-block:: yaml
+
+    language: en
+    pipeline:
+    - name: HFTransformersNLP
+    model_weights: "bert-base-uncased"
+    model_name: "bert"
+    - name: LanguageModelTokenizer
+    - name: LanguageModelFeaturizer
+    - name: DIETClassifier
+    epochs: 50
+
+In both cases we're training a :ref:`diet-classifier` for combined intent classification and entity recognition
+for 50 epochs but there are a few differences.
+
+In the light configuration we have :ref:`CountVectorsFeaturizer` which creates bag-of-word
+representations for each incoming message(at word and character levels). The heavy configuration replaces it with a
+``BERT`` model inside the pipeline. :ref:`HFTransformersNLP` is a utility component that does the heavy lifting work of loading the
+``BERT`` model in memory. Under the hood it leverages HuggingFace's `Transformers library <https://huggingface.co/transformers/>`_ to initialize the specified language model.
+Notice that we add two additional components :ref:`LanguageModelTokenizer` and :ref:`LanguageModelFeaturizer` which
+pick up the tokens and feature vectors respectively that are constructed by the utility component.
+
```

Comment:
```suggestion
In both cases, the features extracted from the text (either a bag-of-words or BERT embeddings) are passed to the DIET model to train.
```

Reply:
It'd be nice to have a proper link to huggingface somewhere tho.